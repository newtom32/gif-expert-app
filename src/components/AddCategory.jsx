import { useState } from 'react'


export const AddCategory = ({onNewCategory}) => {// Es la prop que paso en el propType del compoenente.
  
    const [inputValue, setinputValue] = useState([])

    const onInputChange = (event) => {
        setinputValue(event.target.value);
        
    }

    const onSubmit = (event) =>{
        event.preventDefault();
        if( inputValue.trim().length <= 1) return;

       //setCategories((categories) =>[ inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setinputValue('');
        
        
    }

    return (

        <form onSubmit={ (event) => onSubmit(event)}>
            <input
                type='text'
                placeholder="Buscar gif"
                value={inputValue}
                onChange={ (event) => onInputChange(event)}
            />
        </form>

  )
}



