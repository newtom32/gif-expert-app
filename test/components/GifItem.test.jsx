import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas GifItem', () => {

    const title = 'jhon';
    const url = 'https://jhon.com/';

    test('debe de hacer match con el snapshop', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });


    test('Debe mostrar la imagen con el url y alt indicado', () => {
        
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByRole('img').src).toBe(url);

    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});