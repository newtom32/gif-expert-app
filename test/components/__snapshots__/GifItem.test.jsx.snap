// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifItem debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="jhon"
      src="https://jhon.com/"
    />
    <p>
      jhon
    </p>
  </div>
</div>
`;
